app/models/miq_worker/runner.rb

```diff
@@ -298,15 +296,6 @@ def heartbeat
     do_exit("Error heartbeating because #{err.class.name}: #{err.message}\n#{err.backtrace.join('\n')}", 1)
   end
 
-  def process_messages_from_server
-    worker_monitor_drb.register_worker(@worker.pid, @worker.class.name, @worker.queue_name)
```

Comment:
I think we may still need this call.

This is what adds the worker info to the server hash for fetching queue messages.

Reply:
Should that live in the queue_worker then?  I don't think non-queue workers need that

Reply:
That's part of what it does. This also ensures that the worker is present in the server's worker's cache. It's not all that straight forward unfortunately, but now that the messages bit is gone we can unravel it in a follow-up, but for this first pass I think we should probably just keep this around.

Reply:
Okay, that means that run_single_worker will still fail if the miq_server isn't running (aka testing out a single worker) can we try to handle that case also?